src/js/popup/popup.js

```diff
@@ -338,76 +312,79 @@ async function showRelayPanel(tipPanelToShow) {
 
   //Check if premium features are available
   const premiumCountryAvailability = (await browser.storage.local.get("premiumCountries"))?.premiumCountries?.PREMIUM_PLANS;
- 
+
   //Check if user is premium
   const { premium } = await browser.storage.local.get("premium");
   
   //Check if user has a subdomain set
   const { premiumSubdomainSet } = await browser.storage.local.get("premiumSubdomainSet");
 
   //Educational Panel
-  const educationalModule = premiumPanelWrapper.querySelector(".educational-component");
   const educationalImgEl = premiumPanelWrapper.querySelector(".education-img");
-  const attachmentSizeLimitHeadline = premiumPanelWrapper.querySelector(".education-headline");
-  const attachmentSizeLimitBody = premiumPanelWrapper.querySelector(".education-body");
+  const educationHeadlineEl = premiumPanelWrapper.querySelector(".education-headline");
+  const educationBodyEl = premiumPanelWrapper.querySelector(".education-body");
   const currentEducationalPanel = premiumPanelWrapper.querySelector(".current-panel");
-  const panelPremiumPagination = educationalModule.querySelector(".onboarding-pagination");
-
-  //Load first announcement item
-  const educationStringsSelection = educationalStrings["educationalCriticalEmails"];
-  const educationalComponentStrings = educationStringsSelection;
-  attachmentSizeLimitHeadline.textContent = educationalComponentStrings.headline;
-  attachmentSizeLimitBody.textContent = educationalComponentStrings.description;
-  educationalImgEl.src = educationalComponentStrings.img;
-  currentEducationalPanel.textContent = `${tipPanelToShow}`;
-  educationalModule.setAttribute("id", "educationalCriticalEmails");
-  
-  if (!browser.menus) {
-    panelPremiumPagination.classList.add("hidden");
-  }
-
-  const updateEducationPanel = async (announcementIndex) => {
-    currentEducationalPanel.textContent = [`${tipPanelToShow}`];
-    if (announcementIndex === 1) {
-      switchEducationPanel("educationalCriticalEmails");
-      // educationalModule.classList.remove("is-last-panel");
+  const educationalCtaEl = premiumPanelWrapper.querySelector(".onboarding-cta");
 
-      if (!browser.menus) {
-        // Override class for Chrome browsers to not display sign-back in
-        educationalModule.classList.add("is-last-panel");
-      }
+  const updatePremiumPanel = async (panelId) => {
+    const panelToShow =  `panel${panelId}`;
+    premiumPanelWrapper.setAttribute("id", panelToShow);
+    const panelStrings = premiumPanelStrings.announcements[`${panelToShow}`];
+    if (!panelStrings) {
+      // Exit early if on a non-onboarding
+      return;
+    }
+    educationBodyEl.classList.remove("small-font-size");
+    if (panelStrings.longText) {
+      educationBodyEl.classList.add("small-font-size");
     }
```

Comment:
Reduces font size particularly for longer body texts in the announcement to prevent overlap between elements.

Reply:
In the future, I'd like to impose a character limit of ~100 for premium announcement descriptions.